src/lib/realtime.ts

```diff
@@ -20,14 +20,19 @@ export type SocketWithNoteId = Socket & { noteId: string }
 
 const chance = new Chance()
 
+export enum State {
+  Starting,
+  Running,
+  Stopping
+}
 /* eslint-disable @typescript-eslint/no-use-before-define */
 const realtime: {
   onAuthorizeSuccess: (data, accept) => void;
   onAuthorizeFail: (data, message, error, accept) => void;
   io: SocketIO.Server; isReady: () => boolean;
   connection: (socket: SocketWithNoteId) => void;
   secure: (socket: SocketIO.Socket, next: (err?: Error) => void) => void;
-  getStatus: (callback) => void; maintenance: boolean;
+  getStatus: (callback) => void; state: State;
```

Comment:
Can you please check if this has an impact on the Frontend? If I recall correct the frontend uses maintenance for some sort of state change. (Like making the note read-only.)

Reply:
This should be fine. We don't send the `realtime` object to the frontend, but an explicit `maintenance` message, which we still do. It's in the `handleTermSignal` function.